flutter_svg ^0.17.4 is causing ios build to fail

The flutter_svg plugin version ^0.17.4 causes the ios build to fail. Using `flutter build ios --no-codesign`, error below.

It seems this is not an issue when using flutter_svg ^0.18.0 as I found [here](https://github.com/dnfield/flutter_svg/issues/368)

Is there any objection to bumping the version @oscarmanderj ?
*Change log can be found [here](https://pub.dev/packages/flutter_svg/changelog)*

```
Running Xcode build...                                                  
                                                   
Xcode build done.                                           32.9s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:88:26: Error: Type
    'DiagnosticableMixin' not found.
    class PictureStream with DiagnosticableMixin {
                             ^^^^^^^^^^^^^^^^^^^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:192:44: Error: Type
    'DiagnosticableMixin' not found.
    abstract class PictureStreamCompleter with DiagnosticableMixin {
                                               ^^^^^^^^^^^^^^^^^^^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:88:7: Error: The type
    'DiagnosticableMixin' can't be mixed in.
    class PictureStream with DiagnosticableMixin {
          ^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:192:16: Error: The type
    'DiagnosticableMixin' can't be mixed in.
    abstract class PictureStreamCompleter with DiagnosticableMixin {
                   ^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:167:11: Error: Superclass has
    no method named 'debugFillProperties'.
        super.debugFillProperties(properties);
              ^^^^^^^^^^^^^^^^^^^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:171:30: Error: The method
    'toStringShort' isn't defined for the class 'PictureStreamCompleter'.
     - 'PictureStreamCompleter' is from 'package:flutter_svg/src/picture_stream.dart'
     ('../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'toStringShort'.
          ifPresent: _completer?.toStringShort(),
                                 ^^^^^^^^^^^^^
    ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:266:11: Error: Superclass has
    no method named 'debugFillProperties'.
        super.debugFillProperties(description);
              ^^^^^^^^^^^^^^^^^^^

    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Encountered error while building for device.
```

Yes go ahead and bump it

On Mon, 31 Aug 2020, 12:34 am Connor, <notifications@github.com> wrote:

> The flutter_svg plugin version ^0.17.4 causes the ios build to fail. Using flutter
> build ios --no-codesign, error below.
>
> It seems this is not an issue when using flutter_svg ^0.18.0 as I found
> here <https://github.com/dnfield/flutter_svg/issues/368>
>
> Is there any objction to bumping the version @oscarmanderj
> <https://github.com/oscarmanderj> ?
>
> Running Xcode build...
>
>
>
> Xcode build done.                                           32.9s
>
> Failed to build iOS app
>
> Error output from Xcode build:
>
> ↳
>
>     ** BUILD FAILED **
>
>
>
>
>
> Xcode's output:
>
> ↳
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:88:26: Error: Type
>
>     'DiagnosticableMixin' not found.
>
>     class PictureStream with DiagnosticableMixin {
>
>                              ^^^^^^^^^^^^^^^^^^^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:192:44: Error: Type
>
>     'DiagnosticableMixin' not found.
>
>     abstract class PictureStreamCompleter with DiagnosticableMixin {
>
>                                                ^^^^^^^^^^^^^^^^^^^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:88:7: Error: The type
>
>     'DiagnosticableMixin' can't be mixed in.
>
>     class PictureStream with DiagnosticableMixin {
>
>           ^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:192:16: Error: The type
>
>     'DiagnosticableMixin' can't be mixed in.
>
>     abstract class PictureStreamCompleter with DiagnosticableMixin {
>
>                    ^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:167:11: Error: Superclass has
>
>     no method named 'debugFillProperties'.
>
>         super.debugFillProperties(properties);
>
>               ^^^^^^^^^^^^^^^^^^^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:171:30: Error: The method
>
>     'toStringShort' isn't defined for the class 'PictureStreamCompleter'.
>
>      - 'PictureStreamCompleter' is from 'package:flutter_svg/src/picture_stream.dart'
>
>      ('../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart').
>
>     Try correcting the name to the name of an existing method, or defining a method named 'toStringShort'.
>
>           ifPresent: _completer?.toStringShort(),
>
>                                  ^^^^^^^^^^^^^
>
>     ../../development/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.17.4/lib/src/picture_stream.dart:266:11: Error: Superclass has
>
>     no method named 'debugFillProperties'.
>
>         super.debugFillProperties(description);
>
>               ^^^^^^^^^^^^^^^^^^^
>
>
>
>     Command PhaseScriptExecution failed with a nonzero exit code
>
>     note: Using new build system
>
>     note: Building targets in parallel
>
>     note: Planning build
>
>     note: Constructing build description
>
>
>
> Encountered error while building for device.
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/voteflux/voting_app/issues/137>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF2VQSKFPFQC3VBUFAHTPNLSDJPP5ANCNFSM4QPU5W4A>
> .
>

--- a/.spelling
+++ b/.spelling
@@ -61,6 +61,7 @@ runnercodesignsudo
 ci
 chromelib
 goalConfigurer.ts.
+goalCreator.tslib
  - docs/CODE_OF_CONDUCT.md
 socio-economic
 behavior

--- a/.spelling
+++ b/.spelling
@@ -59,6 +59,8 @@ flushbar
 codesign
 runnercodesignsudo
 ci
+chromelib
+goalConfigurer.ts.
  - docs/CODE_OF_CONDUCT.md
 socio-economic
 behavior

--- a/README.md
+++ b/README.md
@@ -176,27 +176,26 @@ To run the web build: `flutter run -d chrome`
 
 ## CICD
 
-* GitHub --(webhook)--> GitLab CE (cicd project)
-  - mirror clone github.com/voteflux/voting_app
-  - push to gitlab repo
-* GitLab CE (main project) -- cicd triggers normally
-  - runs .gitlab-ci.yaml pipelines
-  - reports status back to github
-  - multiple runners based on tags to deliniate signing / permissions / branch-filters / etc.
-* Avoid multiline scripts in gitlab-ci.yml; use named scripts under `./ci/` instead.
-* coverage is gathered in gitlab but I'm not sure where to find the reports :/, anyway getting covg feedback in PRs would be really nice
-* currently this runs for every push which is not ideal; would be nice to just to PRs
-  * the problem with that is the mirroring situation with gitlab and detecting when PRs are being run vs normal pushes -- gitlab webhook URLs might be able to help us here (combined with event filters on the github side for outgoing webhooks)
+Currently we're using Atomist to trigger CICD jobs which are run on some of Max's machines.
+
+The CICD runner is here: @XertroV/sdm-everything
+
+The particular code that is run on each runner is specified in index.ts (controlled via environment vars): https://github.com/XertroV/sdm-everything/blob/master/index.ts
+
+The exact commands that are run for each flutter job are specified in `lib/goals/goalCreator.ts` - https://github.com/XertroV/sdm-everything/blob/master/lib/goals/goalCreator.ts
+
+> **Note:** Additional config can be found in `lib/goals/goalConfigurer.ts` including some (partially working) caching stuff, and hooks to github statuses, etc.
 
 ### macos codesigning
 
 > **Note: substantial criticism and improvements very welcome**
 
-The codesigning setup is roughly:
+The codesigning setup (specified in [goalCreator.ts](https://github.com/XertroV/sdm-everything/blob/master/lib/goals/goalCreator.ts)) is roughly:
 
-* a gitlab runner instance running under user `runner`
-* a `codesign` user with xcode set up and things
-* very restricted `sudo -u codesign` access for runner
+* build via xcode on CLI
+* sign via xcode on CLI (this part is somewhat troublesome)
+
+There are some sorta-functional scripts Max has developed but they're not all working. You can find them in `./ci`, i.e. here: https://github.com/voteflux/voting_app/tree/master/ci
 
 Problems:
 

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -41,7 +41,7 @@ dependencies:
   dots_indicator: ^1.0.0
   convert: ^2.1.1
   fl_chart: ^0.10.0
-  flutter_svg: ^0.17.4
+  flutter_svg: ^0.18.1
 
 dependency_overrides:
   analyzer: "0.39.14"

